deeprank2/query.py

```diff
@@ -551,161 +551,30 @@ def _load_ppi_pssms(pssm_paths: Optional[Dict[str, str]],
             with open(pssm_path, "rt", encoding="utf-8") as f:
                 chain.pssm = parse_pssm(f, chain)
 
+@dataclass(kw_only=True)
+class ProteinProteinInterfaceQuery(DeepRankQuery):
+    """A query that builds a protein-protein interface graph."""
 
-class ProteinProteinInterfaceAtomicQuery(DeepRankQuery):
-
-    def __init__(  # pylint: disable=too-many-arguments
-        self,
-        pdb_path: str,
-        chain_id1: str,
-        chain_id2: str,
-        pssm_paths: Optional[Dict[str, str]] = None,
-        distance_cutoff: Optional[float] = 5.5,
-        targets: Optional[Dict[str, float]] = None,
-        suppress_pssm_errors: bool = False,
-    ):
-        """
-        A query that builds atom-based graphs, using the residues at a protein-protein interface.
-
-        Args:
-            pdb_path (str): The path to the .PDB file.
-            chain_id1 (str): The .PDB chain identifier of the first protein of interest.
-            chain_id2 (str): The .PDB chain identifier of the second protein of interest.
-            pssm_paths (Optional[Dict(str,str)], optional): The paths to the .PSSM files, per chain identifier. Defaults to None.
-            distance_cutoff (Optional[float], optional): Max distance in Ångström between two interacting atoms of the two proteins.
-                Defaults to 5.5.
-            targets (Optional[Dict(str,float)], optional): Named target values associated with this query. Defaults to None.
-            suppress_pssm_errors (bool, optional): Suppress error raised if .pssm files do not match .pdb files and throw warning instead.
-                Defaults to False.
-        """
-
-        model_id = os.path.splitext(os.path.basename(pdb_path))[0]
-
-        DeepRankQuery.__init__(self, model_id, targets, suppress_pssm_errors)
-
-        self._pdb_path = pdb_path
-
-        self._chain_id1 = chain_id1
-        self._chain_id2 = chain_id2
-
-        self._pssm_paths = pssm_paths
+    def __post_init__(self):
+        super().__post_init__()
+        if len(self.chain_ids) != 2:
```

Comment:
I think the previous logic is more flexible and we shouldn't change it, i.e., giving the possibility to explicitly indicate the two chains involved. Even if a structure has more than 2 chains, it should be possible to specify which are the chains to be considered for the interface.

Reply:
I think, but again will need to check, that the `chain_ids` argument is used 
- by SRVs: to determine which chain the variant is on
- by PPIs: to determine which 2 chains form the interface
This does not prohibit other chains from being part of the graph. I will check the documentation and make sure that this is not misleading.

Reply:
Makes sense, and now the docs are clearer :) I think I was a bit confused here but looking at it now seems good